Assets/BossRoom/Scripts/Client/UI/UnityServicesUIHandler.cs

```diff
@@ -27,21 +27,24 @@ void ServiceErrorHandler(UnityServiceErrorMessage error)
             var errorMessage = error.Message;
             if (error.AffectedService == UnityServiceErrorMessage.Service.Lobby)
             {
-                if ((error.OriginalException as LobbyServiceException).Reason == LobbyExceptionReason.LobbyConflict)
+                switch ((error.OriginalException as LobbyServiceException).Reason)
                 {
-                    // LobbyConflict can have multiple causes. Let's add other solutions here if there's other situations that arise for this.
-                    errorMessage += "\nSee logs for possible causes and solution.";
-                    Debug.LogError($"Got service error {error.Message} with LobbyConflict. Possible conflict cause: Trying to play with two builds on the " +
-                                   $"same machine. Please use command line arg '{ClientMainMenuState.AuthProfileCommandLineArg} someName' to set a different auth profile.\n");
+                    case LobbyExceptionReason.LobbyConflict:
```

Comment:
@fernando-cortez Since you just recently created this, do you think extending this here like I did is the right way to go or would you suggest we do something else?

Reply:
I believe this was @SamuelBellomo's work. However, this looks good.

So if this class is meant to catch all Unity Services errors, I think we'd want to use this class as well to display Authentication errors, no?

`ClientMainMenu` creates a popup message when authentication fails. However, it seems that an authentication message is published inside of `AuthenticationServiceFacade`, but there's no listener for that to create a popup message. Could that popup message creation be moved here instead?

Reply:
I'm thinking we should have a lobby specific message handler and then a catch all for anything unhandled.

Reply:
By lobby specific message handler you'd still mean something in the same class right (just trying to make sure I'd understand this correctly).

So if we were to add a `default` case to the switch here which would show a popup like it did before I manipulated the code here we would be fine. That was an oversight from my side my goal was not to change the initial behaviour so I'll add that back. I think splitting it into a separate class would be quite messy because we'd still need to track here which messages don't need the generic popup.

Reply:
my worry is we end up with a god class here.
My thinking was we'd have one handler per service plus one "catch everything else that falls through the filter" class.
This could be preoptimization though, I'll let you decide.
And yes to the default :)

Reply:
Just a heads up @LukeStampfli that this seems to end up what @pdeschain did [here](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/pull/547).

I think we'll just need to add the lobby full message to the lobby case, correct?

Reply:
Yes I just refactored this and integrated it into @pdeschain's changes.

Reply:
My concerns here were resolved. @SamuelBellomo I'll let you resolve conversation.